src/refda/adm/ietf_dtnma_agent.c

```diff
@@ -1333,13 +1382,183 @@ static void refda_adm_ietf_dtnma_agent_oper_negate(refda_oper_eval_ctx_t *ctx)
      * |START CUSTOM FUNCTION refda_adm_ietf_dtnma_agent_oper_negate BODY
      * +-------------------------------------------------------------------------+
      */
+    const cace_ari_t *val    = refda_oper_eval_ctx_get_operand_index(ctx, 0);
+    cace_ari_t        result = CACE_ARI_INIT_UNDEFINED;
+    int               retval = 0;
+    switch (val->as_lit.prim_type)
+    {
+        case CACE_ARI_PRIM_UINT64:
+            result.as_lit.value.as_uint64 = val->as_lit.value.as_uint64 * -1;
+            break;
+        case CACE_ARI_PRIM_INT64:
+            result.as_lit.value.as_int64 = val->as_lit.value.as_int64 * -1;
+            break;
+        case CACE_ARI_PRIM_FLOAT64:
+            result.as_lit.value.as_float64 = val->as_lit.value.as_float64 * -1.0;
+            break;
+        default:
+            // leave lit as default undefined
```

Comment:
In the ADM itself, it is good practice to log warnings about why a result is undefined.
This should also go into the Conventions.md in doxygen or maybe elsewhere in docs.

Reply:
Fixed and updated docs.